wandb/sdk/launch/agent2/controllers/k8s.py

```diff
@@ -49,103 +49,20 @@ def __init__(
-        self.logger.info(f"Leased item: {json.dumps(next_item, indent=2)}")
-        return next_item
-
-    async def launch_item(self, item: Job) -> Optional[str]:
-        run_id = await self.launch_item_task(item)
-        self.logger.info(f"Launched item got run_id: {run_id}")
-        return run_id
-
-    async def launch_item_task(self, item: Job) -> Optional[str]:
-        self.logger.info(f"Launching item: {json.dumps(item, indent=2)}")
-
-        project = LaunchProject.from_spec(item.run_spec, self.legacy.api)
-        project.queue_name = self.config["jobset_spec"].name
-        project.queue_entity = self.config["jobset_spec"].entity_name
-        project.run_queue_item_id = item.id
-        project.fetch_and_validate_project()
-        run_id = project.run_id
-        job_tracker = self.legacy.job_tracker_factory(run_id)
-        job_tracker.update_run_info(project)
-
-        ack_result = await self.jobset.ack_job(item.id, run_id)
-        if not ack_result:
-            self.logger.error(f"Failed to ack item: {item.id}")
-            return None
-        self.logger.info(f"Acked item: {json.dumps(ack_result, indent=2)}")
-
-        image_uri = None
-        if not project.docker_image:
-            entrypoint = project.get_single_entry_point()
-            assert entrypoint is not None
-            image_uri = await self.legacy.builder.build_image(
-                project, entrypoint, job_tracker
-            )
-        else:
-            assert project.docker_image is not None
-            image_uri = project.docker_image
-
-        assert image_uri is not None
-        run = await self.legacy.runner.run(project, image_uri)
-        if not run:
-            job_tracker.failed_to_start = True
-            self.logger.error(f"Failed to start run for item {item.id}")
-            raise NotImplementedError("TODO: handle this case")
-
-        self.active_runs[item.id] = run
-        self.logger.info(f"Inside launch_item_task, project.run_id = {run_id}")
-        return project.run_id
-
-    async def release_item(self, item_id: str) -> None:
-        self.logger.info(f"Releasing item: {item_id}")
-        del self.active_runs[item_id]
+        super().__init__(config, jobset, logger, legacy, max_concurrency)
+        # TODO: handle orphaned jobs in resource and assign to self (can do
+        # this because we will tell users they can only have one to one
+        # relationships of agents and jobs to queues in a cluster)
+
+    async def find_orphaned_jobs(self) -> List[Dict[str, Any]]:
+        raise NotImplementedError
+
+    async def label_job(self, project: LaunchProject) -> None:
+        k8s_block = await self._get_resource_block(project)
+        if k8s_block is None:
+            return
+        jobset_label = await self._construct_jobset_discoverability_label()
+        k8s_block.get("metadata", {}).get("labels", {})[
+            WANDB_JOBSET_DISCOVERABILITY_LABEL
+        ] = jobset_label
```

Comment:
This won't actually set anything if the "metadata" or "labels" key are missing. You'll have to do something more explicit like:

```
metadata = k8s_block.get("metadata", {})
labels = metadata.get("labels", {})
labels[WANDB_JOBSET_DISCOVERABILITY_LABEL] = jobset_label
metadata["labels"] = labels
k8s_block["metadata"] = metadata
```